objects/src/transaction/consumed_notes.rs

```diff
@@ -199,3 +202,47 @@ impl From<&RecordedNote> for ConsumedNoteInfo {
         Self::new(recorded_note.note().nullifier(), recorded_note.note().script().hash())
     }
 }
+
+// SERIALIZATION
+// ================================================================================================
+//
+
+impl Serializable for ConsumedNotes {
+    fn write_into<W: ByteWriter>(&self, target: &mut W) {
+        target.write_u64(self.notes.len() as u64);
```

Comment:
We impose a limit on the total number of notes which can be consumed in a transaction, and this limit should fit into `u16` - but we should also enforce this limit in the constructors.

Reply:
added assert